### ISSUE TITLE
Allow table.vstack with tables that have coordinate mixin columns

### ISSUE THREAD

**Issue by hamogu** [astropy/astropy#10278]
### Description

I was trying to stack two tables with observations information (XMM and Chandra data in my case). Each table already has a column with a coordinate mixin column, but I can't stack them:
```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-273-9fe785878676> in <module>
----> 1 all_data = table.vstack([chandra_data, xmm_data])

/nfs/melkor/d1/guenther/soft/anaconda/envs/ciao412/lib/python3.7/site-packages/astropy/table/operations.py in vstack(tables, join_type, metadata_conflicts)
    290     col_name_map = OrderedDict()
    291 
--> 292     out = _vstack(tables, join_type, col_name_map, metadata_conflicts)
    293 
    294     # Merge table metadata

/nfs/melkor/d1/guenther/soft/anaconda/envs/ciao412/lib/python3.7/site-packages/astropy/table/operations.py in _vstack(arrays, join_type, col_name_map, metadata_conflicts)
    854         if not hasattr(col_cls.info, 'new_like'):
    855             raise NotImplementedError('vstack unavailable for mixin column type(s): {}'
--> 856                                       .format(col_cls.__name__))
    857         try:
    858             out[out_name] = col_cls.info.new_like(cols, n_rows, metadata_conflicts, out_name)

NotImplementedError: vstack unavailable for mixin column type(s): SkyCoord

```

This is not a critical feature because I can work around it easily (just delete the mixin column, stack, recreate the mixin column from RA/DEC information that I have in other columns) but it would be nice to have. I see that stacking columns might not be easy (e.g. they could use different reference systems), but maybe stacking could work at least in the simplest (and probably most common) case where the frames are identical.

**Comment by bsipocz** [comment]
Support for this added very recently. Have you tried the latest master version?

**Comment by taldcroft** [comment]
This is definitely fixed by #9857.
